source/base/mpi_remote_point_evaluation.cc

```diff
@@ -185,6 +185,36 @@ namespace Utilities
 
 
 
+    template <int dim, int spacedim>
+    bool
+    RemotePointEvaluation<dim, spacedim>::all_points_found() const
+    {
+      if (is_map_unique())
+        return true;
+
+      for (unsigned int i = 1; i < point_ptrs.size(); ++i)
+        if ((point_ptrs[i] - point_ptrs[i - 1]) == 0)
+          return false;
```

Comment:
Do we want to avoid this code redundancy? I would consider it exceptional that points are not found, so the performance advantage by returning earlier before the last element has been reached should maybe not be most relevant. The present function could call `valid_point_masks()` and then apply `std::all_of` (with the argument that the memory required for the vector should be rather small compared to the memory for storing the points, the result data, etc.). If all of this is too expensive, a private member function should be defined that returns whether point i has been found. This member function is then called for both `all_points_found()` and `valid_point_masks()`.